### PR TITLE
Manual backport of Improvement- Added helm inputs for managing audit logs into release/1.0.x

### PR DESCRIPTION
**Pull request by Ganeshrockz**
## Backport

This PR is manually created from #2265 to be assessed for backporting targeting the release/1.0.x

The below text is copied from the body of the original PR.

---

Changes proposed in this PR:
- Adds the relevant inputs in values.yaml to configure audit logging in Consul enterprise.
- Behind the scenes, values get translated to the json configuration present [here](https://developer.hashicorp.com/consul/docs/agent/config/config-files#audit) and the whole configuration is added as a file in the Consul server's config map.

How I've tested this PR:
- Unit tests
- CI

How I expect reviewers to test this PR:
- I've added the `auditLog` field under `server`. Should the same be added to `client`?

Checklist:
- [X] Tests added
- [X] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

---

### CODE REVIEW

**Comment by david-yu** [comment]
This might be related to pipeline issues that @wilkermichael raised on Slack, the same for the 0.49.x PR could be said.

**Comment by david-yu** [comment]
Could you re-based this PR and try running tests again?

File: .changelog/2368.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
```
Comment (curtbushko) [inline .changelog/2368.txt:1]: Same as the other backport, this should use the original PR .changelog/2265.txt

**Review by curtbushko — CHANGES_REQUESTED** [review]

**Review by curtbushko — APPROVED** [review]

**Comment by curtbushko** [comment]
See: https://github.com/hashicorp/consul-k8s/pull/2434

**Comment by david-yu** [comment]
@curtbushko Could you merge if all looks good?

**Pull request merged**